tools/project_manager/page_compare.php

```diff
@@ -0,0 +1,213 @@
+<?php
+$relPath="./../../pinc/";
+include_once($relPath.'base.inc');
+include_once($relPath.'misc.inc');
+include_once($relPath.'theme.inc');
+include_once($relPath.'Project.inc');
+include_once($relPath."DPage.inc"); // remove_formatting()
+
+require_login();
+
+$comparator = new Comparator();
+$comparator->get_data();
+$comparator->render();
+
+class Comparator
+{
+    function __construct()
+    {
+        global $PROJECT_STATES_IN_ORDER;
+
+        $this->L_round_options = array('P1', 'P2', 'P3', 'F1');
+        $this->R_round_options = array('F1', 'F2');
+        $this->state_index = array_flip($PROJECT_STATES_IN_ORDER);
+    }
+
+    function get_data()
+    {
+        $this->projectid = array_get($_GET, 'project', null);
```

Comment:
I prefer you still use `validate_projectID('project', @$_GET['project']);` so it matches the pattern elsewhere in the code. Yes this has the problem that we output an error message before we start the page, but I'd prefer we be consistent as much as possible. In the future perhaps we'll change how we do this, but this should never happen unless someone is mucking with URLs in which case it's their fault they are getting invalid HTML returned from the server.